src/lib/helpers/string.ts

```diff
@@ -53,3 +53,10 @@ export function formatNum(number: number): string {
  * Returns a regex to check hostname validity. Supports wildcards too!
  */
 export const hostnameRegex = String.raw`(\*)|(\*\.)?(?!-)[A-Za-z0-9\-]+([\-\.]{1}[a-z0-9]+)*\.[A-Za-z]{2,18}|localhost`;
+
+/**
+ * Returns a regex to check hostname validity.
+ *
+ * Supports domains, localhost, wildcards, ip-addresses and Chrome extension IDs!
+ */
+export const extendedHostnameRegex = String.raw`(\*)|(\*\.)?((?!-)[A-Za-z0-9-]+([-.]{1}[a-z0-9]+)*\.[A-Za-z]{2,18}|localhost|(\d{1,3}\.){3}\d{1,3}|[a-z0-9]{32})`;
```

Comment:
hmmm this regex allows including spaces 🧐 
<img width="1269" alt="image" src="https://github.com/user-attachments/assets/4a674cc0-9af4-466e-82a2-4b214f28e952" />

Reply:
Ah the pattern is invalid: 
<img width="1555" alt="image" src="https://github.com/user-attachments/assets/31190c30-8530-4f45-add6-c79887345c19" />

Reply:
Some things need to be escaped:

```suggestion
export const extendedHostnameRegex = String.raw`(\*)|(\*\.)?((?!-)[A-Za-z0-9\-]+([\-\.]{1}[a-z0-9]+)*\.[A-Za-z]{2,18}|localhost|(\d{1,3}\.){3}\d{1,3}|[a-z0-9]{32})`;
```